tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -3346,3 +3346,146 @@ def test_13_cleanup(self):
+                self.exception = "Create full snapshot"
+                raise Exception(str(self.exception))
+
+            self.snapshot_ids.append(self.snapshot_id_full)
+
+            LOG.debug("Snapshot ID-1: " + str(self.snapshot_ids[0]))
+            reporting.add_test_step("Full Snapshot is created.",
+                                    tvaultconf.PASS)
+
+
+            # Create incremental-1 snapshot
+            self.snapshot_id_inc = self.workload_snapshot(
+                workload_id, False)
+            self.wait_for_workload_tobe_available(workload_id)
+            if (self.getSnapshotStatus(workload_id, self.snapshot_id_inc) != "available"):
+                self.exception = "Create incremental-1 snapshot"
+                raise Exception(str(self.exception))
+
+            self.snapshot_ids.append(self.snapshot_id_inc)
+            LOG.debug("Snapshot ID-2: " + str(self.snapshot_ids[1]))
+            reporting.add_test_step("Incremental Snapshot is created.",
+                                    tvaultconf.PASS)
+
+            volume_snapshots1 = self.get_volume_snapshots(volume_id)
+            LOG.debug("Volumes snapshots for: " +
+                      str(volume_id) + ": " + str(volume_snapshots1))
+            if len(volume_snapshots1) == 1:
+                reporting.add_test_step("Unencrypted Volume snapshot count is 1", tvaultconf.PASS)
+            else:
+                reporting.add_test_step("Unencrypted Volume snapshot count is not 1", tvaultconf.FAIL)
+                raise Exception("Unencrypted Volume snapshot count is not 1")
+
+            volume_snapshots2 = self.get_volume_snapshots(encryp_volume_id)
+            LOG.debug("Volumes snapshots for: " +
+                      str(encryp_volume_id) + ": " + str(volume_snapshots2))
+            if len(volume_snapshots2) == 1:
+                reporting.add_test_step("Encrypted Volume snapshot count is 1", tvaultconf.PASS)
+            else:
+                reporting.add_test_step("Encrypted Volume snapshot count is not 1", tvaultconf.FAIL)
+                raise Exception("Encrypted Volume snapshot count is not 1")
+
+            self.workload_reset(workload_id)
+
+            start_time = time.time()
+            time.sleep(10)
+            volsnaps_after = self.get_volume_snapshots(volume_id)
+            while (len(volsnaps_after) != 0 and (time.time() - start_time < 600)):
+                volsnaps_after = self.get_volume_snapshots(volume_id)
+                time.sleep(5)
+            if len(volsnaps_after) == 0:
+                reporting.add_test_step("Temp Volume snapshot is deleted after workload reset", tvaultconf.PASS)
+            else:
+                LOG.debug("Workload reset failed")
+                reporting.add_test_step("Temp Volume snapshot not deleted after workload reset", tvaultconf.FAIL)
+                raise Exception("Workload reset failed as temp volume snapshot is not deleted")
```

Comment:
we can mark test_step as fail in except block if we are raising exception.

Reply:
In except block it marks test case as failed and here it defines which step failed